usr/src/cmd/zdb/zdb.c

```diff
@@ -3164,8 +3164,8 @@ zdb_dump_block(char *label, void *buf, uint64_t size, int flags)
 	for (i = 0; i < nwords; i += 2) {
 		(void) printf("%06llx:  %016llx  %016llx  ",
 		    (u_longlong_t)(i * sizeof (uint64_t)),
-		    (u_longlong_t)(do_bswap ? BSWAP_64(d[i]) : d[i]),
-		    (u_longlong_t)(do_bswap ? BSWAP_64(d[i + 1]) : d[i + 1]));
+		    (u_longlong_t)BE_64(d[i]),
+		    (u_longlong_t)BE_64(d[i + 1]));
```

Comment:
@asomers @ahrens 
The point of the do_bswap is for switching the representation of the number on screen. So you need to swap the 64 bit number accordingly. As for the header, it's for indicating the real byte sequence, 0 needs to be the first byte, f needs to be the last byte. So the header needs to swap with the number.

The current code does this correctly on big endian system, but not on little endian system as you can see from the example in my patch.